accepted/0000-missing-command-lookups.md

```diff
@@ -0,0 +1,70 @@
+# Support Yarn style command/script/bin lookups from the CLI
+
+## Summary
+
+Allow for a CLI experience like:
+
+```sh
+npm install typescript
+npm tsc
+```
+
+## Motivation
+
+I have two main motivations:
+
+- Personal, looking at the list of yarn commands I [use in an average month](https://gist.github.com/orta/489dbfa1b7467804955ea7800adf1494#file-last_month_yarn-txt) (1200) roughly 30% (400) of those commands are using the shorthand syntax to run a command installed into `node_modules/.bin` in a project.
+
+- Professional, it'd be great for the TypeScript docs to be able to simply say: `npm install typescript --save-dev' then `npm tsc index.ts` and there to be no ambiguity like our current recommendation for `npx tsc index.ts`.
+
+## Detailed Explanation
+
+When npm receives a command it does not recognise, npm first resolves the next string to see if it matches:
+
+ - Available "scripts" in `package.json`
+ - Commands matching in `./node_modules/.bin/`
+
+Failing that, then show help.
+
+## Rationale and Alternatives
+
+1. Having a `script`s section like this:
+
+    ```json
+    {
+      "scripts": {
+        "tsc": "tsc",
+        "eslint": "eslint",
+        "prettier": "prettier"
+      }
+    }
+    ```
+
+    Is a waste of space in the JSON, the `node_modules/.bin` folder is typically full of commands and after a while you either start prefixing commands with `node_modules/.bin` instead or you fill up the scripts section.
+
+2. This change removes the concept of "blessed" scripts. `npm test` works but `npm test:lint` does not. Eventually folks memorize the ones they can use shorthand for, but there's no reason for the distinction other than at some point it was decided to be given a short-cut. These special cases can be removed with this change.
+
+3. It does not break any existing setups. 
+ 
+ - `npm tsc` does not work today, anyone who is doing that is seeing a help message.
+ - `npm test:lint` also does not work today.
+
+
+Alternative to:
+
+- [RFC 279:for --default-command](https://github.com/npm/rfcs/pull/279) - tries to hit the same issue, but because it'd a user option it wouldn't be useful at the eco-system level and only people who know it exists would use it.
+
+## Implementation
+
+I'm not sure on why so many npm commands on npm require the `--`.
```

Comment:
this is a normal unix convention to tell one command to stop processing `--` commands and pass the rest on untouched; imo people are unfamiliar with it largely because a few popular tools have opted to try to hide that convention from users.

Reply:
Heh, yeah, this bit was my notes before I wrote below - dropped it in the current version